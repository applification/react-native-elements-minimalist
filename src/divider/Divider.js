import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-primitives';

const Divider = ({ fullWidth, color }) => (
  <View
    style={{
      backgroundColor: color,
      marginTop: 10,
      marginLeft: fullWidth ? 0 : 30,
      height: 1,
      width: '100%',
    }}
  />
);

Divider.propTypes = {
  fullWidth: PropTypes.bool.isRequired,
  color: PropTypes.string.isRequired,
};

Divider.defaultProps = {
  fullWidth: false,
  color: '#DADADA',
};

export default Divider;
