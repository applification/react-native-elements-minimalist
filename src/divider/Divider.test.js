import React from 'react';
import renderer from 'react-test-renderer';

import Divider from './Divider';

describe('DIVIDER', () => {
  it('Renders with DefaultProps', () => {
    const component = renderer.create(<Divider />).toJSON();
    expect(component.props.style.marginLeft).toBe(30);
    expect(component).toMatchSnapshot();
  });

  it('Renders with color prop', () => {
    const component = renderer.create(<Divider color="red" />).toJSON();
    expect(component.props.style.backgroundColor).toBe('red');
    expect(component).toMatchSnapshot();
  });

  it('Renders with fullWidth prop', () => {
    const component = renderer.create(<Divider fullWidth />).toJSON();
    expect(component.props.style.width).toBe('100%');
    expect(component.props.style.marginLeft).toBe(0);
    expect(component).toMatchSnapshot();
  });
});
