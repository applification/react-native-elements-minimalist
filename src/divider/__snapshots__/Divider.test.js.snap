// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DIVIDER Renders with DefaultProps 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#DADADA",
      "height": 1,
      "marginLeft": 30,
      "marginTop": 10,
      "width": "100%",
    }
  }
/>
`;

exports[`DIVIDER Renders with color prop 1`] = `
<View
  style={
    Object {
      "backgroundColor": "red",
      "height": 1,
      "marginLeft": 30,
      "marginTop": 10,
      "width": "100%",
    }
  }
/>
`;

exports[`DIVIDER Renders with fullWidth prop 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#DADADA",
      "height": 1,
      "marginLeft": 0,
      "marginTop": 10,
      "width": "100%",
    }
  }
/>
`;
